tesseract_ros/tesseract_monitoring/src/contact_monitor.cpp

```diff
@@ -37,45 +39,87 @@ static DiscreteContactManager::Ptr manager;
 static ros::Subscriber joint_states_sub;
 static ros::Publisher contact_results_pub;
 static ros::Publisher environment_pub;
+static ros::Publisher contact_marker_pub;
 static ros::Subscriber environment_diff_sub;
 static ros::ServiceServer modify_env_service;
 static ros::ServiceServer compute_contact_results;
 static ContactTestType type;
 static ContactResultMap contacts;
 static tesseract_msgs::ContactResultVector contacts_msg;
 static bool publish_environment;
+static bool publish_markers;
 static boost::mutex modify_mutex;
 static DiscreteContactManagerPluginLoaderPtr discrete_manager_loader; /**< The discrete contact manager loader */
+static std::vector<std::string> monitor_link_names;
+static double contact_distance;
+static boost::shared_ptr<sensor_msgs::JointState> current_joint_states;
+static boost::condition_variable current_joint_states_evt;
 
-void callbackJointState(const sensor_msgs::JointState::ConstPtr& msg)
+void computeCollisionReportThread()
 {
-  boost::mutex::scoped_lock(modify_mutex);
-  contacts.clear();
-  contacts_msg.contacts.clear();
-
-  tess->getEnvironment()->setState(msg->name, msg->position);
-  EnvState::ConstPtr state = tess->getEnvironment()->getCurrentState();
-
-  manager->setCollisionObjectsTransform(state->transforms);
-  manager->contactTest(contacts, type);
-
-  if (publish_environment)
+  while (!ros::isShuttingDown())
   {
-    tesseract_msgs::TesseractState state_msg;
-    toMsg(state_msg, *(tess->getEnvironment()));
-    environment_pub.publish(state_msg);
+    boost::mutex::scoped_lock lock(modify_mutex);
```

Comment:
I limited the scope of the lock. Please take a look to verify I did not limit it to much.

Reply:
I think this is fine. Again, in the future it might be nice to have a separate thread/mutex to avoid locking the joint state callback with contactTest, but I think this is good. At least publishing won't lock it.